app/decorators/decorators.py

```diff
@@ -50,11 +51,46 @@ def wrapper(*args, **kwargs):
                     return jsonify({'error': 'Permission denied. Admin or Manager role required.'}), 403
             except Exception as e:
                 return jsonify({'error': f'Error in role check: {e}'}), 500
-        
-
         return wrapper
     return decorator
 
+def token_required(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        auth_header = request.headers.get('Authorization')
+
+        if not auth_header:
+            return jsonify({'message': 'Token is missing!'}), 401
+        parts = auth_header.split()
+
+        if parts[0].lower() != 'bearer':
+            return jsonify({'message': 'Authorization header must start with Bearer'}), 401
+        elif len(parts) == 1:
+            return jsonify({'message': 'Token not found'}), 401
+        elif len(parts) > 2:
+            return jsonify({'message': 'Authorization header must be Bearer + \s + token'}), 401
```

Comment:
All these checking can be done in a single if condition, If you get Authorization default value as a string.

And just split it with Bearer.